packages/reports/src/Constants/data-table-constants.tsx

```diff
@@ -12,7 +12,19 @@ import MarketSymbolIconRow from '../Components/market-symbol-icon-row.jsx';
 import ProfitLossCell from '../Components/profit_loss_cell.jsx';
 import CurrencyWrapper from '../Components/currency-wrapper';
 
-const getModeFromValue = key => {
+type TActionType =
```

Comment:
@akmal-deriv Let's spend some time on refactoring the code to make TS more sense 🙏🏻 

We can `drive` types from the map we already have, we don’t need to duplicate them, Then we can refactor `getModeFromValue` to be more specific on types, Then we can change the type of `cell_value` in `TCellContentProps` to `TKeys` and remove the unnecessary `as` at line `114`, And also remove all the unnecessary typings for `renderCellContent`.

```ts
const map = {
    buy: 'success',
    deposit: 'success',
    hold: 'warn',
    release: 'success',
    sell: 'danger',
    withdrawal: 'info',
    default: 'default',
    adjustment: 'adjustment',
    transfer: 'transfer',
} as const;

type TKeys = keyof typeof map;

const getModeFromValue = (key: TKeys) => map[key] || map.default;
```

Reply:
Thanks, changed👍🏻